diagnostic_feedback/tests/integration/test_student_dg_quiz.py

```diff
@@ -0,0 +1,154 @@
+
+        self.assertEqual('Efficient' in final_result, True)
+
+    def test_for_single_grp_mediocre_team(self):
+        self.load_student_view('dg_quiz', {"mode": "standard"})
+
+        next_btn = self._get_next_button()
+        back_btn = self._get_previous_button()
+        startover_link = self._get_startover_button_link()
+
+        self.assertEqual(startover_link.is_displayed(), False)
+
+        self._verify_question(0, next_btn, back_btn, 0)
+        self._submit_question()
+
+        self._verify_question(1, next_btn, back_btn, 1)
+        self._submit_question()
+
+        self._verify_question(2, next_btn, back_btn, 0)
+        self._submit_question(is_last=True)
+
+        self.wait_until_visible(startover_link)
+
+        self.assertEqual(startover_link.is_displayed(), True)
+
+        final_result = self.browser.find_element_by_css_selector('div.response_body').text
+
+        self.assertEqual('Mediocre' in final_result, True)
+
+    def test_for_single_grp_awful_team(self):
+        self.load_student_view('dg_quiz', {"mode": "standard"})
+
+        next_btn = self._get_next_button()
+        back_btn = self._get_previous_button()
+        startover_link = self._get_startover_button_link()
+
+        self.assertEqual(startover_link.is_displayed(), False)
+
+        self._verify_question(0, next_btn, back_btn, 2)
+        self._submit_question()
+
+        self._verify_question(1, next_btn, back_btn, 1)
+        self._submit_question()
+
+        self._verify_question(2, next_btn, back_btn, 2)
+        self._submit_question(is_last=True)
+
+        self.wait_until_visible(startover_link)
+
+        self.assertEqual(startover_link.is_displayed(), True)
+
+        final_result = self.browser.find_element_by_css_selector('div.response_body').text
+
+        self.assertEqual('Awful' in final_result, True)
+
+    def test_for_multi_group(self):
+        self.load_student_view('dg_multi_group_quiz', {"mode": "standard"})
+
+        next_btn = self._get_next_button()
+        back_btn = self._get_previous_button()
+        startover_link = self._get_startover_button_link()
+
+        self.assertEqual(startover_link.is_displayed(), False)
+
+        self._verify_question(0, next_btn, back_btn, 0)
+        self._submit_question()
+
+        self._verify_question(1, next_btn, back_btn, 0)
+        self._submit_question()
+
+        self._verify_question(2, next_btn, back_btn, 0)
+        self._submit_question()
+
+        self._verify_question(3, next_btn, back_btn, 0)
+        self._submit_question()
+
+        self._verify_question(4, next_btn, back_btn, 1)
+        self._submit_question()
+
+        self._verify_question(5, next_btn, back_btn, 0)
+        self._submit_question(is_last=True)
+
+        self.wait_until_visible(startover_link)
+
+        self.assertEqual(startover_link.is_displayed(), True)
+
+        final_results = self.browser.find_elements_by_css_selector('div.response_body div.result')
```

Comment:
@jareerahsan This could also get its own method (`DiagnosticFeedbackBaseTest._get_final_results`).